src/components/FormField/FormFieldSelect.js

```diff
@@ -1,36 +1,74 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import Select from 'react-select';
+import { useField } from 'formik';
 import styles from './FormField.module.scss';
+import mainStyles from '~/assets/styles/main';
+import FormFieldErrorBoundary from '../error-boundaries/FormFieldErrorBoundary';
+import variables from '~/assets/styles/variables';
+
+const mapValues = (values) => (values ? values.map((val) => ({ label: val, value: val })) : []);
 
 const FormFieldSelect = ({
-  label, options, onChange, value,
-}) => (
-  <div className={styles.FormField}>
-    <label>{label}</label>
-    <Select
-      className={styles.selectWrapper}
-      options={options}
-      isMulti
-      onChange={onChange}
-      value={value}
-    />
-  </div>
-);
+  label, options, name, placeholder,
+}) => {
+  const [, meta, helpers] = useField(name);
+  const { value: selectedValues } = meta;
+  const { setValue, setTouched } = helpers;
+
+  const onSelectChange = (newValues) => {
+    setValue(newValues && newValues.length > 0 ? newValues.map(({ value }) => value) : null);
+  };
+
+  const isInvalid = meta.touched && meta.error;
+
+  const customStyles = {
+    control: (base) => (isInvalid
+      ? {
+        ...base,
+        borderColor: variables.errorBorderColor,
+        '&:hover': {
+          borderColor: variables.errorBorderColor,
+        },
+        backgroundColor: variables.errorBackgroundColor,
+      }
+      : { ...base }),
```

Comment:
it is a bit hard to read. Maybe smth like this will be better?
```js
control: (base) => {
  const errorStyles = {
    borderColor: variables.errorBorderColor,
    '&:hover': {
        borderColor: variables.errorBorderColor,
    },
    backgroundColor: variables.errorBackgroundColor,
  }

  return isInvalid ? { ...base, ...errorStyles } : base
}
```

Reply:
Thanks, refactored